src/__test__/Badge.spec.js

```diff
@@ -5,44 +5,54 @@ beforeEach(cleanup);
 
 describe('Badge', () => {
   test('should render text and default color', () => {
-    const { container } = render(Badge, {
+    const { queryByRole } = render(Badge, {
       props: { children: 'Hello world!' }
     });
-    const badge = container.querySelector('.badge');
-    expect(badge.innerHTML).toBe('Hello world!');
-    expect(badge.className).toBe('badge bg-secondary');
+    const badge = queryByRole('status');
```

Comment:
Hi @DaniAcu , thanks, but I'm not sure if this is the right direction for testing here, as Sveltestrap is fundamentally a Bootstrap component framework, which is all based on following their class names.  

I'm also reluctant to hard code all roles for each component, as some users may want to pass in the role as needed.

Does this make sense?  It's certainly an interesting consideration.

Reply:
Maybe hardcode is not the way to go. However, I think that as UI library we should provide more than just a wrapper of styles.
If we are creating a component this peace have one responsibility, or one main use case.
As a UI library it should help to our users to create better sites, adding accessibility and semantics by default could be a really good plus to do it. 
I agree with you that we should be continue be open to any use case, but we have to create a default use case contemplating accessibility and semantics.
Alerts and Badges are good cases to use roles. Obviously for more generic component like the layout it doesn't matter.
What do you think? Could we move to this approach?